### PR TITLE
[BasicUI] Implement Buttongrid widget

### PR DESCRIPTION
**Pull request by lolodomo**
Support grids containing from 1 to 12 columns.
Only the first 8 columns are rendered on tablet.
Only the first 4 columns are rendered on phone.

Related to openhab/openhab-core#3441

Signed-off-by: Laurent Garnier <lg.hc@free.fr>

### CODE REVIEW

**Comment by lolodomo** [comment]
I choose the option to have a fixed width for each button. Look at the capture in the preceding message, it is now really not bad at all.
I have just a very small problem of alignment to solve.
I will also mask some columns when displaying on a phone or on a tablet. I would like to have 4 columns on a phone and 8 on a tablet.

**Comment by lolodomo** [comment]
@kaikreuzer : I am now satisfied with the result. I tested with grids containing from 1 to 12 columns and I tested on Firefox mainly but also quickly with Edge and Chrome. I believe it is ready for review and an interesting new feature for OH 4.1 ;)

**Comment by lolodomo** [comment]
@kaikreuzer : please do not forget me. Maybe you can have a look to the capture screenshots first?

File: bundles/org.openhab.ui.basic/src/main/java/org/openhab/ui/basic/internal/render/SwitchRenderer.java [hunk 1]
```diff
@@ -138,7 +138,6 @@ public EList<Widget> renderWidget(Widget w, StringBuilder sb, String sitemap) th
                 }
             }
             snippet = snippet.replace("%buttons%", buttons.toString());
-            snippet = snippet.replace("%count%", Integer.toString(nbButtons));
```
Comment (kaikreuzer) [inline bundles/org.openhab.ui.basic/src/main/java/org/openhab/ui/basic/internal/render/SwitchRenderer.java:141]: What does this change in the Switch renderer do?

Comment (lolodomo) [reply bundles/org.openhab.ui.basic/src/main/java/org/openhab/ui/basic/internal/render/SwitchRenderer.java:141]: Nothing, it was not used, so I removed it. Just cleaning.

**Review by kaikreuzer — APPROVED** [review]
Thanks!

**Comment by lolodomo** [comment]
Excellent, thank you. A new feature to be mentioned in the final release notes ;)

**Pull request merged**